blocks/user-licenses/user-licenses.js

```diff
@@ -1,5 +1,352 @@
+    if (containerDiv !== null && containerDiv !== undefined) {
+      modalChangeUserContentDiv = containerDiv;
+    }
+  });
+
+  // clear previous modal content
+  // TODO merge with above function
+  modalChangeUserContentDiv.innerHTML = '';
+  modalChangeUserContentDiv.appendChild(await createColleaguesTable('change-user-checkbox'));
+  const userChangeCheckboxes = document.querySelectorAll('.change-user-checkbox');
+  userChangeCheckboxes.forEach((userChangeCheckbox) => {
+    userChangeCheckbox.addEventListener('click', updateChangeUserId);
+  });
+}
+
+async function removeUserFromLicense(event) {
+  const newproductuserid = event.target.getAttribute('data-new-productuserid');
+  const data = await execute('dynamics_remove_enduser_from_license', `&new_productuserid=${newproductuserid}`, 'DELETE');
+  // TODO handle response and add toast message
+
+  if (data !== undefined && data !== null && data.status === 204) {
+    // TODO show success toast message
+    hideModal(event);
+    // eslint-disable-next-line no-use-before-define
+    displayLicenseDetails(event);
+  } else {
+    console.log('error ', data);
+  }
+}
+
+async function updateLicenseNickname() {
+  const { value } = document.querySelector('.nickname');
+  if (value !== '' && value !== undefined) {
+    const data = await execute('dynamics_set_license_nickname', `&id=97d53652-122f-e611-80ea-005056831cd4&nickname=${value}`, 'PATCH');
+    console.log('Updated license nickname ', data);
+  } else {
+    // TODO handle response and add toast message
+    alert('Provide nickname value before saving');
+  }
+}
+
+function hideOtherUserLicenseInformation() {
+  const sections = document.querySelectorAll('.section');
+
+  sections.forEach((section) => {
+    if (section.classList.contains('user-licenses-container')) {
+      const tabsContent = section.querySelector('.tabs');
+      if (tabsContent !== undefined && tabsContent !== null) {
+        section.querySelector('.tabs').remove();
+      }
+      const licenseComponent = section.querySelector('.user-licenses.block');
+      licenseComponent.querySelector('h3').parentElement.parentElement.remove();
+
+      let backToProfileButton = licenseComponent.querySelector('#backToAccountButton');
+      if (backToProfileButton === undefined || backToProfileButton === null) {
+        backToProfileButton = createTag('a', { href: '/pages/profile', class: 'button primary', id: 'backToAccountButton' }, 'Back to Account');
+        licenseComponent.insertBefore(backToProfileButton, licenseComponent.firstChild);
+      }
+    } else {
+      section.remove();
+    }
+  });
+}
+
+async function displayLicenseDetails(event) {
+  hideOtherUserLicenseInformation();
+  const licenseId = event.target.getAttribute('data-license-id');
+  const userId = localStorage.getItem('auth0_id');
+  const accessToken = localStorage.getItem('accessToken');
+
+  if (userId == null || userId === undefined || accessToken == null || accessToken === undefined) {
```

Comment:
try not to compare with `null` and `undefined` directly. See https://developer.mozilla.org/en-US/docs/Glossary/Truthy